src/xmipp/libraries/reconstruction/resolution_fso.cpp

```diff
@@ -0,0 +1,1876 @@
+    	MAT_ELEM(angles, 0, 79) = 44.867209;	 MAT_ELEM(angles, 1, 79) = -75.219088;
+    	MAT_ELEM(angles, 0, 80) = 27.132791;	 MAT_ELEM(angles, 1, 80) = -75.219088;
+    	}
+
+    }
+
+    void interpolationCoarse(MultidimArray< double > fsc,
+    		const Matrix2D<double> &angles,
+			Matrix1D<double> &freq_fourier_x,
+			Matrix1D<double> &freq_fourier_y,
+			Matrix1D<double> &freq_fourier_z,
+    		MultidimArray<double> &threeD_FSC,
+			MultidimArray<double> &counterMap,
+			MultidimArray< double >& freqMap,
+			MultidimArray< double >& freq,
+			double maxFreq, int m1sizeX, int m1sizeY, int m1sizeZ,
+			double rot, double tilt, double ang_con)
+    {
+		int ZdimFT1=(int)ZSIZE(threeD_FSC);
+		int YdimFT1=(int)YSIZE(threeD_FSC);
+		int XdimFT1=(int)XSIZE(threeD_FSC);
+
+		double x_dir, y_dir, z_dir, uz, uy, ux, cosAngle, aux;
+		cosAngle = cos(ang_con);
+		x_dir = sin(tilt*PI/180)*cos(rot*PI/180);
+		y_dir = sin(tilt*PI/180)*sin(rot*PI/180);
+		z_dir = cos(tilt*PI/180);
+//		aux = 1.0/pow(ang_con,6);
+		aux = 4.0/((cos(ang_con) -1)*(cos(ang_con) -1));
+		long n = 0;
+		for (int k=0; k<ZdimFT1; k++)
+		{
+			double uz = VEC_ELEM(freq_fourier_z,k);
+			uz *= z_dir;
+			for (int i=0; i<YdimFT1; i++)
+			{
+				double uy = VEC_ELEM(freq_fourier_y,i);
+				uy *= y_dir;
+				for (int j=0; j<XdimFT1; j++)
+				{
+					double ux = VEC_ELEM(freq_fourier_x,j);
+					ux *= x_dir;
+					double iun = DIRECT_MULTIDIM_ELEM(freqMap, n);
+					double f = 1/iun;
+					iun *= (ux + uy + uz);
+					double cosine = fabs(iun);
+
+					if (cosine>=cosAngle)
+						{
+						if (f>maxFreq)
+						{
+							++n;
+							continue;
+						}
+						int idx = (int) round(f * m1sizeX);
+						cosine = exp( -((cosine -1)*(cosine -1))*aux );
+						DIRECT_MULTIDIM_ELEM(threeD_FSC, n) += cosine*dAi(fsc, idx);
+						DIRECT_MULTIDIM_ELEM(counterMap, n) += cosine;
+						}
+					++n;
+				}
+			}
+		}
+    }
+
+    //TODO: Merge with Simple
+    void anistropyParameter(const MultidimArray<double> FSC,
+    		MultidimArray<double> &directionAnisotropy, size_t dirnumber,
+			MultidimArray<double> &aniParam, double thrs)
+    {
+    	double N = 0;
+		for (size_t k = 0; k<aniParam.nzyxdim; k++)
+		{
+			if (DIRECT_MULTIDIM_ELEM(FSC, k) >= thrs)
+			{
+				DIRECT_MULTIDIM_ELEM(aniParam, k) += 1.0;
+				N++;
+			}
+		}
+		DIRECT_MULTIDIM_ELEM(directionAnisotropy, dirnumber) = N;
+    }
+
+    void anistropyParameterSimple(const MultidimArray<double> FSC,
+			MultidimArray<double> &aniParam, double thrs)
+    {
+
+		for (size_t k = 0; k<aniParam.nzyxdim; k++)
+			if (DIRECT_MULTIDIM_ELEM(FSC, k) >= thrs)
+				DIRECT_MULTIDIM_ELEM(aniParam, k) += 1.0;
+    }
+
+    void prepareData(FileName &fnhalf1, FileName &fnhalf2,
+    		MultidimArray<double> &half1, MultidimArray<double> &half2, bool test)
+    {
+    	MultidimArray<double> &phalf1 = half1, &phalf2 = half2;
+
+    	Image<double> mask;
+    	MultidimArray<double> &pmask = mask();
+
+    	if (test)
```

Comment:
tests should be separately

Reply:
I cannot do a test right now for this algorithm. This test is defined to develop the algorithm. But starting with google test, if a lot of time I do not have before the release...